### PR TITLE
send encoded password and UI changes

### PR DESCRIPTION
**Pull request by hemantkumargangwal**

### CODE REVIEW

File: apps/proxy-auth/src/app/otp/component/register/register.component.ts [hunk 1]
```diff
@@ -77,7 +81,21 @@ export class RegisterComponent extends BaseComponent implements AfterViewInit, O
                 }
             });
     }
-
+    ngOnChanges(changes: SimpleChanges) {
+        if (changes.prefillDetails && changes.prefillDetails.currentValue) {
```
Comment (rahul1choudhary) [inline apps/proxy-auth/src/app/otp/component/register/register.component.ts:85]:
```suggestion
        if (changes?.prefillDetails?.currentValue) {
```

File: apps/proxy-auth/src/app/otp/component/login/login.component.html [hunk 2]
```diff
@@ -88,17 +131,61 @@
...
-            type: 'password',
-            label: 'New Password',
-            formControl: resetPasswordForm.get('password'),
-            patternError:
-                'Password should contain atleast one Capital Letter, one Small Letter, one Digit and one Symbol'
-        }"
+
+    <mat-form-field
+        appearance="outline"
+        style="width: 100%; padding-bottom: 4px"
```
Comment (rakeshshrivas) [inline apps/proxy-auth/src/app/otp/component/login/login.component.html:137]: Convert into class

File: apps/proxy-auth/src/app/otp/component/register/register.component.ts [hunk 3]
```diff
@@ -41,7 +44,7 @@ export class RegisterComponent extends BaseComponent implements AfterViewInit, O
                 Validators.maxLength(25),
             ]),
             email: new FormControl<string>(null, [Validators.required, Validators.pattern(EMAIL_REGEX)]),
-            mobile: new FormControl<string>(null, [Validators.required]),
+            mobile: new FormControl<any>(null, [Validators.required]),
```
Comment (rahul1choudhary) [inline apps/proxy-auth/src/app/otp/component/register/register.component.ts:47]:
```suggestion
            mobile: new FormControl<string>(null, [Validators.required]),
```

File: apps/proxy-auth/src/app/otp/component/login/login.component.html [hunk 4]
```diff
@@ -15,29 +15,72 @@
...
+        <mat-error *ngIf="userNameControl.errors?.required"> Email or Mobile number is required.</mat-error>
+        <mat-hint class="note">
+            <span class="subheading"> Note:</span> Please enter your Mobile number with the country code (e.g.
+            91)</mat-hint
+        >
+    </mat-form-field>
+
+    <mat-form-field
+        appearance="outline"
+        style="width: 100%; margin-top: 16px"
```
Comment (rakeshshrivas) [inline apps/proxy-auth/src/app/otp/component/login/login.component.html:30]: Convert into class

File: apps/proxy-auth/src/app/otp/component/login/login.store.ts [hunk 5]
```diff
@@ -26,13 +27,15 @@ export class LoginComponentStore extends ComponentStore<ILoginInitialState> {
...
             showRegistration: null,
+            prefillDetails: null,
         });
     }
     readonly otpdata$: Observable<any> = this.select((state) => state.otpData);
     readonly isLoading$: Observable<boolean> = this.select((state) => state.isLoading);
     readonly resetPassword$: Observable<IResetPassword> = this.select((state) => state.resetPassword);
     readonly apiError$: Observable<any> = this.select((state) => state.apiError);
     readonly showRegistration$: Observable<boolean> = this.select((state) => state.showRegistration);
+    readonly prefillDetails$: Observable<string> = this.select((state) => state.prefillDetails);
```
Comment (rahul1choudhary) [inline apps/proxy-auth/src/app/otp/component/login/login.store.ts:38]: remove this

File: apps/proxy-auth/src/app/otp/component/login/login.component.ts [hunk 6]
```diff
@@ -85,7 +93,13 @@ export class LoginComponent extends BaseComponent implements OnInit {
         });
         this.componentStore.showRegistration$.pipe(filter(Boolean), takeUntil(this.destroy$)).subscribe((res) => {
             if (res) {
-                this.showRegistration();
+                this.prefillDetails = this.loginForm.get('username').value;
+                this.showRegistration(this.prefillDetails);
+            }
+        });
+        this.componentStore.prefillDetails$.pipe(takeUntil(this.destroy$)).subscribe((res) => {
```
Comment (rahul1choudhary) [inline apps/proxy-auth/src/app/otp/component/login/login.component.ts:100]: remove this

File: apps/proxy-auth/src/app/otp/component/login/login.component.html [hunk 7]
```diff
@@ -88,17 +131,61 @@
...
+        <mat-label>Password</mat-label>
+        <input
+            matInput
+            [type]="showPassword ? 'text' : 'password'"
+            placeholder="Password"
+            autocomplete="off"
+            [formControl]="passwordControl"
+        />
+
+        <button matSuffix mat-icon-button (click)="showPassword = !showPassword">
```
Comment (rahul1choudhary) [inline apps/proxy-auth/src/app/otp/component/login/login.component.html:149]: reuse this code

**Review by rahul1choudhary — CHANGES_REQUESTED** [review]

**Review by shubhamkeshari — APPROVED** [review]

**Pull request closed without merging**